# START-UPS PROFIT PREDICTIONS

 This is a Machine Learning project to predict the profits of some startup companies using the linear regression technique.

In [1]:
#Importing libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
#importing our dataset
dataset = pd.read_csv('C:/Users/opeye/OneDrive/Documents/Python Data Analyst/50_Startups.csv')


In [3]:
#Convert categorical variables into numerical features by creating a dummy variables
states = pd.get_dummies(dataset['State'], drop_first=True)

In [4]:
#Create our features and target
X = dataset.iloc[:,:4]
y = dataset.iloc[:,4]

In [5]:
#Drop the state column and concatenate the new column with the X dataset
X = X.drop('State', axis=True)
X = pd.concat([X, states], axis=1)
y.head(3)

0    192261.83
1    191792.06
2    191050.39
Name: Profit, dtype: float64

In [6]:
X.head()

R&D Spend  Administration  Marketing Spend  Florida  New York
0  165349.20       136897.80        471784.10        0         1
1  162597.70       151377.59        443898.53        0         0
2  153441.51       101145.55        407934.54        1         0
3  144372.41       118671.85        383199.62        0         1
4  142107.34        91391.77        366168.42        1         0

In [7]:
#Split the dataset into training and testing set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test=train_test_split(X,y,test_size=0.3,random_state=33)

In [8]:
#Fit in the model in the Training set
from sklearn.linear_model import LinearRegression
regr = LinearRegression()
regr.fit(X_train, y_train)

LinearRegression()

In [9]:
#Make the prediction
y_pred = regr.predict(X_test)

In [10]:
#Check accuracy of model
from sklearn.metrics import r2_score
score = r2_score(y_test, y_pred)
score

0.9256662855582558

In [11]:
#Coefficients of our linear regression model
coef = regr.coef_
coef

array([ 8.46076819e-01, -7.55086154e-02,  3.61732601e-02,  5.97477071e+02,
        3.11629672e+03])

The result showed that the model has a 92.6% accuracy for unseen data. This means the model had a high performance on unseen data, which can be confidently used for future predictions.